lib/st/api/lims/ml_warehouse.pm

```diff
@@ -32,6 +33,39 @@ in WTSI::DNAP::Warehouse::Schema.
 
 =head2 id_flowcell_lims
 
+=head2 id_run
+
+id_run, optional attribute.
+
+=cut
+has '+id_run' =>       ( required        => 0,
+                         lazy_build      => 1,
+                       );
+
+sub _build_id_run {
+  my $self = shift;
+  if (not $self->has_flowcell_barcode and not $self->has_id_flowcell_lims) {
+    croak 'Require flowcell_barcode or id_flowcell_lims to try to find id_run';
+  }
+  my%search;
+  if($self->has_flowcell_barcode){
+    $search{'flowcell_barcode'} = $self->flowcell_barcode;
+  }
+  if($self->has_id_flowcell_lims){
```

Comment:
The same logic as in query_resultset() in ml_warehouse. Should it be factored out into a public function in ml_warehouse?